ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -111,9 +112,11 @@ public JoinNode(
         Optional.empty()
     );
 
-    this.schema = buildJoinSchema(joinKey, left, right);
     this.joinType = requireNonNull(joinType, "joinType");
     this.joinKey = requireNonNull(joinKey, "joinKey");
+    this.schema = joinKey.isForeignKey()
```

Comment:
I think we need to refactor this later: If the purpose of `JoinKey` is to define the key of the result, it should not know anything about PK vs FK joins. (Guess it's good enough for now, to get the PR merged).